scripts/release/mule/test/test.sh

```diff
@@ -3,19 +3,33 @@
 
 set -ex
 
+echo
+date "+build_release begin TEST stage %Y%m%d_%H%M%S"
+echo
+
+if [ -z "$BRANCH" ] || [ -z "$NETWORK" ] || [ -z "$SHA" ]; then
+    echo "[$0] BRANCH=$BRANCH, NETWORK=$NETWORK or SHA=$SHA is missing."
+    exit 1
+fi
+
+CHANNEL=$(./scripts/release/mule/common/get_channel.sh "$NETWORK")
+export CHANNEL
+
+VERSION=${VERSION:-$(./scripts/compute_build_number.sh -f)}
+export VERSION
+
+export BRANCH
+
 export PKG_TYPE="$1"
+
 ARCH_BIT=$(uname -m)
 export ARCH_BIT
-ARCH_TYPE=$(./scripts/archtype.sh)
-export ARCH_TYPE
-OS_TYPE=$(./scripts/ostype.sh)
+
+OS_TYPE=$(./scripts/release/mule/common/ostype.sh)
 export OS_TYPE
 
-export BRANCH=${BRANCH:-$(./scripts/compute_branch.sh)}
-export CHANNEL=${CHANNEL:-$(./scripts/compute_branch_channel.sh "$BRANCH")}
-export NETWORK=${NETWORK:-$(./scripts/compute_branch_network.sh "$BRANCH")}
-export SHA=${SHA:-$(git rev-parse HEAD)}
-export VERSION=${VERSION:-$(./scripts/compute_build_number.sh -f)}
+export SHA
```

Comment:
I'm debating on whether nor not this should be auto-determined. I understand why we would require it, but it is an extra manual step.

Reply:
Yeah, if we could improve this it would be great.  I'm not sure how we could do it where we'd always be sure to get the correct hash.